engine/access/ping/engine.go

```diff
@@ -107,18 +107,25 @@ func (e *Engine) startPing() {
 // pingNode pings the given peer and updates the metrics with the result and the additional node information
 func (e *Engine) pingNode(peer *flow.Identity) {
 	id := peer.ID()
+	pingFailed := false
 
 	// ping the node
 	resp, rtt, err := e.middleware.Ping(id) // ping will timeout in libp2p.PingTimeoutSecs seconds
 	if err != nil {
 		e.log.Debug().Err(err).Str("target", id.String()).Msg("failed to ping")
 		// report the rtt duration as negative to make it easier to distinguish between pingable and non-pingable nodes
 		rtt = -1
+		pingFailed = true
 	}
 
 	// get the additional info about the node
 	info := e.nodeInfo[id]
 
 	// update metric
-	e.metrics.NodeReachable(peer, info, rtt, resp.Version, resp.BlockHeight)
+	e.metrics.NodeReachable(peer, info, rtt)
+
+	// if ping succeeded then update the node info metric
+	if !pingFailed {
```

Comment:
alternative is to rename `err` to `pingErr`, then just doing if `pingErr` == nil, which i think gives the same result with a little more clarity, or include it in an `else` for the if statement on line 114.